modules/server/src/network/aggregations/AggregationAccumulator.ts

```diff
@@ -43,14 +44,14 @@ const resolveAggregations = ({
  * @param type
  * @param aggregations
  */
-export const resolveToNetworkAggregation = (
-	type: SupportedAggregation,
-	aggregations: Aggregations[],
-): AllAggregations | undefined => {
+const resolveToNetworkAggregation = (
+	type: string,
+	aggregations: Aggregations[] | NumericAggregations[],
+): Aggregations | NumericAggregations => {
 	if (type === SUPPORTED_AGGREGATIONS.Aggregations) {
-		return resolveAggregation(aggregations);
+		return resolveAggregation(aggregations as Aggregations[]);
 	} else if (type === SUPPORTED_AGGREGATIONS.NumericAggregations) {
-		return resolveNumericAggregation(aggregations);
+		return resolveNumericAggregation(aggregations as NumericAggregations);
```

Comment:
Wondering how we could make better type checks here rather than using 'as'

Reply:
I think there's likely a more optimal solution for the TS.

I feel comfortable about this usage of `as` because if you look at the conditional, at runtime we are checking that it is of a type, it's just not informing the typechecker correctly.